src/winc1500/src/winc.c

```diff
@@ -82,6 +82,14 @@ static void resolve_callback(uint8_t *host, uint32_t ip)
  */
 static void socket_callback(SOCKET sock, uint8_t msg_type, void *msg)
 {
+    // Close Socket on SEND/SENDTO Error.
```

Comment:
Why isn't this in the switch below in line 160 ?

Reply:
Also why is it needed ? The socket will be closed when send()/sendto() returns an error.

Reply:
I want it to run whenever anything happens so that the socket gets closed. This is why I put it there. If you put it in the switch then it would not be run for any socket callback.

I had to add this because the winc module accepts data for any invalid sockets without errors. I.e. when you send. It will accept the data always even if the socket was closed from the other side. The winc module only communicates the socket is bad via the callback.

Reply:
Note, it still blocks you sometimes until there's buffer space... but, it will never just block you indefinitely like you'd expect on a broken socket. The module accepts the data, throws it away, and then returns that it couldn't send the data via the callback.